internal/lsp/server.go

```diff
@@ -971,6 +976,20 @@ func (s *Server) handleInlayHint(
 	return languageService.ProvideInlayHint(ctx, params)
 }
 
+func (s *Server) handleCodeLens(ctx context.Context, ls *ls.LanguageService, params *lsproto.CodeLensParams) (lsproto.CodeLensResponse, error) {
+	return ls.ProvideCodeLenses(ctx, params.TextDocument.Uri)
+}
+
+func (s *Server) handleCodeLensResolve(ctx context.Context, codeLens *lsproto.CodeLens, reqMsg *lsproto.RequestMessage) (*lsproto.CodeLens, error) {
```

Comment:
This is likely to conflict with Sheetal's PR, but probably in a good way given the codelens needs to do cross-project too.

Reply:
I can do that part as a separate PR but strada didn’t do it so I didn’t port it in my earlier pr